### DISCUSSION TITLE
PixiJs canvas shapes disappear after window resizing

### DISCUSSION THREAD

**Discussion by raghavshivi94** [pixijs/pixijs#8148] [Q&A]
I am using PixiJS to render shapes on canvas. When I resize the window(decrease window size) the 'this.containerWidth' becomes 0 at one point and so for a moment canvas container disappears from the view and now when I resize again(increase window size)  the canvas container does not hold the previously rendered shapes anymore.
This is how I am handling the resizing:
````
import { Viewport } from 'pixi-viewport';
import { Renderer } from 'pixi.js', 

private _handleWindowResize = () => {
    this.renderer.resize(this.containerWidth, this.containerHeight);
    this.editorViewport.resize(this.containerWidth, this.containerHeight);
  };
````

How can I retain the shapes on canvas as before resizing?

**Comment by bigtimebuddy** [comment]
Please provide a runnable example

**Comment by lilmart210** [comment]
I was having this problem as well, I believe I boiled it down to this.
React calls render on components when the window size change. This caused my reference to be resetted and external rendering to stall. I still have not figured out a way to avoid the component to be refreshed as memoization, use callback amongst others things have not worked.
